pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/DescriptorLookupCache.java

```diff
@@ -112,11 +112,15 @@ public synchronized StepDescriptor lookupStepDescriptor(String n) {
         return describableMap.get(n);
     }
 
-    public synchronized Descriptor<? extends Describable> lookupStepOrFunction(String name) {
+    public synchronized Descriptor<? extends Describable> lookupStepFirstThenFunction(String name) {
         return lookupStepDescriptor(name) != null ? lookupStepDescriptor(name) : lookupFunction(name);
     }
 
-    public synchronized DescribableModel<? extends Describable> modelForStepOrFunction(String name) {
+    public synchronized Descriptor<? extends Describable> lookupFunctionFirstThenStep(String name) {
```

Comment:
Just realized something not exactly related to this, but we probably need to invalidate the cache when a new plugin is installed or someone configures a new global library.